### DISCUSSION TITLE
Getting Corruption Error on Mac when Trying to run  or install App from DMG

### DISCUSSION THREAD

**Discussion by markdemich** [BowlerHatLLC/vscode-as3mxml#673] [Q&A] [answered]
I created a signed app and I create a dmg to house the app.   If I double click the DMG to open it and then double click the app it all works as expected.  However, if I put the dmg on a cdn and then download it from a browser, when I attempt to either install or run the app inside the DMG it gives me an error saying it's corrupted.

Does anyone know what I need to do to fix this?

**Comment by joshtynjala** [comment, marked as answer]
You need to sign your app with a certificate from Apple and submit it to Apple for notarization.

This guide may be helpful:

https://airsdk.dev/docs/tutorials/platform/macos/creating-macos-pkg-files

**Comment by markdemich** [comment]
Thanks I had searched high and low for something like this. I'll give it a
try.
